usersync/syncersbuilder.go

```diff
@@ -98,28 +101,88 @@ func chooseSyncerConfig(biddersSyncerConfig []namedSyncerConfig) (namedSyncerCon
+			}
 			syncerConfig = bidder
 		}
 	}
 
-	if len(bidderNamesWithEndpoints) == 0 {
+	// check if bidders have same syncer key but no endpoints
+	if len(nonAliasBidderNames)+len(aliasBidderNames) == 0 {
 		sort.Strings(bidderNames)
 		bidders := strings.Join(bidderNames, ", ")
 		return namedSyncerConfig{}, fmt.Errorf("bidders %s share the same syncer key, but none define endpoints (iframe and/or redirect)", bidders)
 	}
 
-	if len(bidderNamesWithEndpoints) > 1 {
-		sort.Strings(bidderNamesWithEndpoints)
-		bidders := strings.Join(bidderNamesWithEndpoints, ", ")
+	// check if non-alias bidders have same syncer key
+	if len(nonAliasBidderNames) > 1 {
+		sort.Strings(nonAliasBidderNames)
+		bidders := strings.Join(nonAliasBidderNames, ", ")
 		return namedSyncerConfig{}, fmt.Errorf("bidders %s define endpoints (iframe and/or redirect) for the same syncer key, but only one bidder is permitted to define endpoints", bidders)
 	}
 
+	if len(parentBidderNameMap) != 0 {
+		// check if alias bidders of different parent have same syncer key
+		if len(parentBidderNameMap) > 1 {
+			sort.Strings(aliasBidderNames)
+			return namedSyncerConfig{}, fmt.Errorf("either alias bidders %s defines endpoints (iframe and/or redirect) for the same syncer key used by multiple bidders, but only one bidder is permitted to define endpoints", strings.Join(aliasBidderNames, ", "))
+		}
+
+		// check if aliases of same parent and non-alias bidder have same syncer key
+		if _, ok := parentBidderNameMap[nonAliasBidderNames[0]]; !ok {
+			sort.Strings(aliasBidderNames)
+			return namedSyncerConfig{}, fmt.Errorf("either alias bidders %s defines endpoints (iframe and/or redirect) for the same syncer key used by other bidder, but only one bidder is permitted to define endpoints", strings.Join(aliasBidderNames, ", "))
+		}
+	}
+
 	return syncerConfig, nil
 }
+
+func getSyncerKey(biddersWithSyncerCfg map[string]config.BidderInfo, bidderName string, bidderInfo config.BidderInfo) (string, error) {
+	if bidderInfo.Syncer == nil {
+		return "", fmt.Errorf("found no syncer config for bidder %s", bidderName)
+	}
+
+	var (
+		bidderSyncerCfg       = bidderInfo.Syncer
+		isAlias               = len(bidderInfo.AliasOf) > 0
+		hasInheritedSyncerCfg = false
+		parentBidderInfo      config.BidderInfo
+		parentBidderName      string
+	)
+
+	if isAlias {
+		parentBidderName = bidderInfo.AliasOf
+		var parentHasSyncerCfg bool
+		parentBidderInfo, parentHasSyncerCfg = biddersWithSyncerCfg[parentBidderName]
+		hasInheritedSyncerCfg = parentHasSyncerCfg && parentBidderInfo.Syncer == bidderSyncerCfg
+	}
+
+	if bidderSyncerCfg.Key == "" {
+		if hasInheritedSyncerCfg {
+			return parentBidderName, nil
+		}
+		return bidderName, nil
+	}
+
+	if isAlias && !hasInheritedSyncerCfg && bidderSyncerCfg.Key == parentBidderInfo.Syncer.Key {
+		return "", fmt.Errorf("syncer key of alias bidder %s is same as the syncer key for its parent bidder %s", bidderName, parentBidderName)
+	}
```

Comment:
That's ok. Should be no harm if an alias specifically sets the parent bidder's key. Is that was this error covers? We can leave the conflict for the other code block to catch.